hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector.java

```diff
@@ -184,44 +185,51 @@ private void scanOperationRunner(long nowNanos, long nowMillis, OperationRunner
             }
 
             // create the stack trace once for this operation and cache the invocation for upcoming updates
-            SlowOperationLog log = getOrCreateSlowOperationLog(operationRunner);
-            operationData.invocation = log.getOrCreateInvocation(operationHashCode, durationNanos, nowNanos, nowMillis);
+            SlowOperationLog log = getOrCreateSlowOperationLogOrNull(operationRunner, operation);
+            if (log != null) {
+                int totalInvocations = log.totalInvocations.incrementAndGet();
+                operationData.invocation = log.getOrCreateInvocation(operationHashCode, durationNanos, nowNanos, nowMillis);
 
-            logSlowOperation(operationRunner, log);
+                logSlowOperation(operation, log, totalInvocations);
+            }
         }
 
-        private SlowOperationLog getOrCreateSlowOperationLog(final OperationRunner operationRunner) {
+        private SlowOperationLog getOrCreateSlowOperationLogOrNull(OperationRunner operationRunner, Object operation) {
```

Comment:
I prevent creating the stack trace string and the log if the operation has changed after the stack trace creation (I cannot control that). In this case I return `null` to signal that the log creation was aborted.

Let me split the method to make this more clear.